docs/docs-beta/docs/guides/external-systems/apis.md

```diff
@@ -1,4 +1,78 @@
 ---
 title: Connecting to APIs
 sidebar_position: 20
----
\ No newline at end of file
+---
+
+When building a data pipeline, you'll likely need to connect to a number of external APIs, each with its own specific configuration and behavior. This guide demonstrates how to standardize your API connections and customize their configuration using Dagster resources.
+
+
+## What you'll learn
+
+- How to connect to an API using a Dagster resource
+- How to use that resource in an asset
+- How to configure a resource
+- How to source configuration values from environment variables
+
+<details>
+  <summary>Prerequisites</summary>
+
+To follow the steps in this guide, you'll need:
+
+- Familiarity with [Asset definitions](/concepts/assets)
+- Familiarity with [resources](/concepts/resources)
+- Install the `requests` library:
+    ```bash
+    pip install requests
+    ```
+
+</details>
+
+## Step 1: Write a resource to connect to an API
+
+Here is a minimal resource for connecting to an API that returns the sunrise time.
+
+In this example, the request URL has been hardcoded to query for data at San Francisco International Airport.
+
+<CodeExample filePath="guides/external-systems/apis/minimal_resource.py" language="python" title="Resource to connect to the Sun API" />
+
+
+## Step 2: Use the resource in an asset
+
+To use the resource written in Step 1, you can provide it to an asset like this:
+
+<CodeExample filePath="guides/external-systems/apis/use_minimal_resource_in_asset.py" language="python" title="Use the SunResource in an asset" />
+
+When you materialize `sfo_sunrise`, Dagster will provide an initialized `SunResource` to the `sun_resource` parameter.
+
+
+## Step 3: Configure your resource
+Many APIs have configuration you can set to customize your usage. Here is an updated version of the resource from Step 1 with configuration to allow for setting the query location:
+
+<CodeExample filePath="guides/external-systems/apis/configurable_resource.py" language="python" title="Configurable Resource to connect to the Sun API" />
```

Comment:
I think you can combine the code from 3 and 3.1 together here to keep it simpler.